src/App.tsx

```diff
@@ -1,26 +1,66 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
+
+import { USER_ID, getTodos } from './api/todos';
+
+import { ErrorNotification } from './components/ErrorNotification';
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Loader } from './components/Loader';
+
+import { DispatchContext, StateContext } from './store/TodoContext';
+import { useErrorMessage } from './components/useErrorMessage';
 
-const USER_ID = 0;
+import { ActionType } from './types/Actions';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const { todos } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+  const handleError = useErrorMessage();
+
+  useEffect(() => {
+    dispatch({ type: ActionType.SetIsInputFocused, payload: true });
```

Comment:
Add helper function for each ActionType in slice file and reuse it here. See example:
```suggestion
const setInputAction = (focused: boolean): SetInputAction => ({
  type: ActionType.SetIsInputFocused,
  payload: focused,
});
```

Reply:
thank you! great suggestion!